edu_edfi_airflow/dags/earthbeam_dag.py

```diff
@@ -501,6 +535,400 @@ def build_tenant_year_taskgroup(self,
+        :param full_refresh:
+
+        :return:
+        """
+        taskgroup_grain = f"{tenant_code}_{api_year}"
+        if grain_update:
+            taskgroup_grain += f"_{grain_update}"
+
+        # Group ID can be defined manually or built dynamically
+        if not group_id:
+            group_id = f"{taskgroup_grain}__earthmover"
+
+            # TaskGroups have three shapes:
+            if edfi_conn_id:         # Earthmover-to-Lightbeam (with optional S3)
+                group_id += "_to_lightbeam"
+            elif snowflake_conn_id:  # Earthmover-to-Snowflake (through S3)
+                group_id += "_to_snowflake"
+            elif s3_conn_id:         # Earthmover-to-S3
+                group_id += "_to_s3"
+
+
+        with TaskGroup(
+            group_id=group_id,
+            prefix_group_id=prefix_group_id,
+            dag=self.dag
+        ) as dynamic_tenant_year_task_group:
+
+            # Dynamically build a task-order as tasks are defined.
+            task_order = []
+            paths_to_clean = []
+
+            ### PythonOperator Preprocess
+            if python_callable:
+                python_preprocess = PythonOperator(
+                    task_id= f"{taskgroup_grain}_{python_task_name}" if python_task_name else f"{taskgroup_grain}_preprocess_python",
+                    python_callable=python_callable,
+                    op_kwargs=python_kwargs or {},
+                    provide_context=True,
+                    pool=self.pool,
+                    dag=self.dag
+                )
+
+                task_order.append(python_preprocess)
+                paths_to_clean.append(airflow_util.xcom_pull_template(python_preprocess.task_id))
+
+            ## List Raw files for dynamic task mapping
+            list_files_task = PythonOperator(
+                task_id = f'{taskgroup_grain}_list_files_in_dir',
+                dag=self.dag,
+                python_callable=self.list_files_in_raw_dir,
+                op_kwargs={
+                    'raw_dir': airflow_util.xcom_pull_template(python_preprocess.task_id) if python_callable else raw_dir,
+                    'ds_nodash_string': '{{ ds_nodash }}',
+                    'ts_nodash_string': '{{ ts_nodash }}'
+                },
+                provide_context=True
+            )
+            task_order.append(list_files_task)
+
+            ### Raw to S3
+            if s3_conn_id:
+                if not s3_filepath:
+                    raise ValueError(
+                        "Argument `s3_filepath` must be defined to upload raw files to S3."
+                    )
+
+                s3_raw_filepath = edfi_api_client.url_join(
+                    s3_filepath, 'raw',
+                    tenant_code, self.run_type, api_year, grain_update,
+                    '{{ ds_nodash }}', '{{ ts_nodash }}'
+                )
+
+                raw_to_s3_kwargs = list_files_task.output.map(lambda raw_dir: {'local_filepath': raw_dir['local_filepath'] })
```

Comment:
@rlittle08 Why not just update this line to the following:
```
raw_to_s3_kwargs = list_files_task.output.map(lambda raw_dir: {
    'local_filepath': raw_dir['local_filepath'],
    's3_conn_id': s3_conn_id,
    's3_destination_key': s3_raw_filepath,
    'remove_local_filepath': False,
})
```

And then remove the need for the `static_kwargs` argument?

Reply:
This might work. I thought I had some issue with passing static values as dynamic args, but I can't remember the details of it, so I think this is worth a shot